ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListSourceExecutor.java

```diff
@@ -39,17 +42,44 @@
 import io.confluent.ksql.util.KsqlException;
 import io.confluent.ksql.util.KsqlStatementException;
 import io.confluent.ksql.util.PersistentQueryMetadata;
+import java.util.LinkedList;
 import java.util.List;
 import java.util.Optional;
 import java.util.function.Predicate;
 import java.util.stream.Collectors;
+import org.apache.kafka.common.KafkaException;
 
 // CHECKSTYLE_RULES.OFF: ClassDataAbstractionCoupling
 public final class ListSourceExecutor {
   // CHECKSTYLE_RULES.ON: ClassDataAbstractionCoupling
 
   private ListSourceExecutor() { }
 
+  private static Optional<KsqlEntity> sourceDescriptionList(
+      final ConfiguredStatement<?> statement,
+      final KsqlExecutionContext executionContext,
+      final ServiceContext serviceContext,
+      final List<? extends StructuredDataSource<?>> sources
```

Comment:
Just use the interface. `StructuredDataSource` is an implementation detail only.

```suggestion
      final List<? extends DataSource<?>> sources
```